report/test/renderer/report-ui-features-test.js

```diff
@@ -117,7 +117,239 @@ describe('ReportUIFeatures', () => {
+          },
+          {
+            'name': 'notexample.com',
+            'company': 'notexample.com',
+            'isUnrecognized': true,
+          },
+        ];
+        lhr.audits['entity-classification'].details.originLUT = {
+          'http://www.example.com': 0,
+          'http://www.cdn.com': 1,
+          'http://www.notexample.com': 2,
+        };
+        lhr.audits['entity-classification'].details.nameLUT = {
+          'example.com': 0,
+          'cdn.com': 1,
+          'notexample.com': 2,
+        };
+
+        // Interleave first/third party URLs to test restoring order.
+        lhr.audits['modern-image-formats'].details.items = [
+          {
+            ...webpAuditItemTemplate,
+            url: 'http://www.cdn.com/img1.jpg', // Third party, will be filtered.
+            entity: 'cdn.com',
+          },
+          {
+            ...webpAuditItemTemplate,
+            url: 'http://www.example.com/img2.jpg', // First party, not filtered.
+            entity: 'example.com',
+          },
+          {
+            ...webpAuditItemTemplate,
+            url: 'http://www.notexample.com/img3.jpg', // Third party, will be filtered.
+            entity: 'notexample.com',
+          },
+        ];
+
+        // Test sub-item rows.
+        lhr.audits['unused-javascript'].details.items = [
+          {
+            ...webpAuditItemTemplate,
+            entity: 'cdn.com',
+            url: 'http://www.cdn.com/script1.js', // Third party, will be filtered.
+            subItems: {
+              type: 'subitems',
+              items: [
+                {source: '1', sourceBytes: 1, sourceWastedBytes: 1},
+                {source: '2', sourceBytes: 2, sourceWastedBytes: 2},
+              ],
+            },
+          },
+          {
+            ...webpAuditItemTemplate,
+            entity: 'example.com',
+            url: 'http://www.example.com/script2.js', // First party, not filtered.
+            subItems: {
+              type: 'subitems',
+              items: [
+                {source: '3', sourceBytes: 3, sourceWastedBytes: 3},
+                {source: '4', sourceBytes: 4, sourceWastedBytes: 4},
+              ],
+            },
+          },
+          {
+            ...webpAuditItemTemplate,
+            entity: 'notexample.com',
+            url: 'http://www.notexample.com/script3.js', // Third party, will be filtered.
+            subItems: {
+              type: 'subitems',
+              items: [
+                {source: '5', sourceBytes: 5, sourceWastedBytes: 5},
+                {source: '6', sourceBytes: 6, sourceWastedBytes: 6},
+              ],
+            },
+          },
+        ];
+
+        // Only third party URLs to test that checkbox is hidden
+        lhr.audits['render-blocking-resources'].details.items = [
+          {
+            ...renderBlockingAuditItemTemplate,
+            entity: 'cdn.com',
+            url: 'http://www.cdn.com/script1.js', // Third party.
```

Comment:
remove `// Third party`, replace with `isFirstParty: false`? Any maybe a single value without any property to test default value being false?

Reply:
Not sure if I follow. 

`isFirstParty` is an attribute in `entity-classification` audit, that's already `false` for this entry (cdn.com) as it's missing (see L131). The comments are copied over from the legacy filtering code below, but they seem good for the reader to distinguish the entries to be filtered, no? I could be misunderstanding your comment.

Reply:
All good, just an idea. I usually try to replace comments with equivalent code if it reads the same; but I don't think it makes much difference here.